src/PKSim.Core/Mappers/PKValuesToPKAnalysisMapper.cs

```diff
@@ -31,15 +31,21 @@ public class PKValuesToPKAnalysisMapper : IPKValuesToPKAnalysisMapper
          _displayUnitRetriever = displayUnitRetriever;
       }
 
-      public PKAnalysis MapFrom(DataColumn dataColumn, PKValues pkValues, PKParameterMode mode, string moleculeName)
+      public PKAnalysis MapFrom(DataColumn dataColumn, PKValues pkValues, PKParameterMode mode, string moleculeName, bool forPopulation)
       {
          var pk = new PKAnalysis().WithName(moleculeName);
-         _pkParameterRepository.All().Where(parameter => parameter.Mode.Is(mode)).Each(parameter => pk.Add(createPKParameter(parameter, pkValues)));
+         _pkParameterRepository.All().Where(parameter => parameter.Mode.Is(mode) && (!forPopulation || !filteNaNValues(parameter.Name, pkValues))).Each(parameter => pk.Add(createPKParameter(parameter, pkValues)));
 
          pk.MolWeight = dataColumn.DataInfo.MolWeight;
          return pk;
       }
 
+      private bool filteNaNValues(string name, PKValues pkValues)
+      {
+         var pkValue = pkValues.ValueFor(name);
+         return (!pkValue.HasValue || float.IsNaN(pkValue.Value));
```

Comment:
hard for me to read without thinking about it 
(!pkValue.HasValue || float.IsNaN(pkValue.Value)); a! and a ||  
Also the method name as a typo (filte). MAybe a better name would be isPKValueDefined so that you do not have to do a ! and the code would be pkValue.HasValue && !float.isNaN(pkValue.value)


&& and ! are, I believe easier to compute than || and ! (in brain power not in computer power)